pydantic/fields.py

```diff
@@ -264,6 +274,14 @@ def validate(self, v, values, *, loc, cls=None):
             v, errors = self._apply_validators(v, values, loc, cls, self.whole_post_validators)
         return v, errors
 
+    def _validate_json(self, v, loc):
+        if self.parse_json:
+            try:
+                return json.loads(v), None
+            except (json.JSONDecodeError, TypeError):
```

Comment:
you can use `ValueError` instead of `son.JSONDecodeError`. Do we actually need `TypeError`?

Reply:
and below.

Reply:
I included `TypeError` in `pydantic/fields.py` to handle these cases:
```>>> json.loads(12)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.6/json/__init__.py", line 348, in loads
    'not {!r}'.format(s.__class__.__name__))
TypeError: the JSON object must be str, bytes or bytearray, not 'int'```

Reply:
ok, but that should raise a descendent if `TypeError` not `ValueError`, best to use a separate `except`